src/datasources/cache/cache.first.data.source.spec.ts

```diff
@@ -35,11 +36,78 @@ describe('CacheFirstDataSource', () => {
     );
   });
 
-  it('should return the data returned by the underlying network interface', async () => {
+  afterEach(() => {
+    jest.useRealTimers();
+  });
+
+  it('should return the data returned by the underlying network interface and cache it', async () => {
+    const targetUrl = faker.internet.url({ appendSlash: false });
+    const cacheDir = new CacheDir(faker.word.sample(), faker.word.sample());
+    const notFoundExpireTimeSeconds = faker.number.int();
+    const data = JSON.parse(fakeJson());
+    mockNetworkService.get.mockImplementation((url) => {
+      switch (url) {
+        case targetUrl:
+          return Promise.resolve({ data });
+        default:
+          return Promise.reject(`No matching rule for url: ${url}`);
+      }
+    });
+
+    const actual = await cacheFirstDataSource.get({
+      cacheDir,
+      url: targetUrl,
+      notFoundExpireTimeSeconds,
+    });
+
+    expect(actual).toEqual(data);
+    expect(mockNetworkService.get).toHaveBeenCalledTimes(1);
+    expect(fakeCacheService.keyCount()).toBe(1); // only data is cached (as no invalidation happened yet a this point in time)
+    expect(await fakeCacheService.get(cacheDir)).toEqual(JSON.stringify(data));
+  });
+
+  it('should return the network data and it should cache it if the last invalidation happened before the request was initiated', async () => {
     const targetUrl = faker.internet.url({ appendSlash: false });
     const cacheDir = new CacheDir(faker.word.sample(), faker.word.sample());
     const notFoundExpireTimeSeconds = faker.number.int();
     const data = JSON.parse(fakeJson());
+    const invalidationTimeMs = jest.now(); // invalidation happens at this point in time
+    fakeCacheService.set(
+      new CacheDir(`invalidationTimeMs:${cacheDir.key}`, ''),
+      invalidationTimeMs.toString(),
+    );
+    mockNetworkService.get.mockImplementation((url) => {
+      switch (url) {
+        case targetUrl:
+          return Promise.resolve({ data });
+        default:
+          return Promise.reject(`No matching rule for url: ${url}`);
+      }
+    });
+
+    jest.advanceTimersByTime(1); // the request is sent 1 ms after invalidation happened
+    const actual = await cacheFirstDataSource.get({
+      cacheDir,
+      url: targetUrl,
+      notFoundExpireTimeSeconds,
+    });
+
+    expect(actual).toEqual(data);
+    expect(mockNetworkService.get).toHaveBeenCalledTimes(1);
+    expect(fakeCacheService.keyCount()).toBe(2); // both data and invalidation timestamp are cached
+    expect(await fakeCacheService.get(cacheDir)).toEqual(JSON.stringify(data)); // item is cached
+  });
+
+  it("should return the network data but it shouldn't not cache it if the last invalidation happened after the request was initiated", async () => {
```

Comment:
```suggestion
  it("should return the network data but it should not cache it if the last invalidation happened after the request was initiated", async () => {
```

Reply:
Agree with your suggestion but when changing `shouldn't` to `should not` we can use single quotes. Change: [b92584c](https://github.com/safe-global/safe-client-gateway/pull/970/commits/b92584cf5c50470b34305c8c9afffcc884c9fb21)

Reply:
(And [01321e7](https://github.com/safe-global/safe-client-gateway/pull/970/commits/01321e7b46fdf3e8914f83348cdb94393babaff1) 😅)